From 3a376e789fb1dde718eda814318a6e4e4d47c1c7 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Mon, 4 Nov 2024 15:16:27 +0100
Subject: [PATCH 11/16] board: solidrun: lx2160acex7: allocate memory before
 patching lx2162 fdt

Update of second usb controller fdt node during board_fix_fdt fails due
to lack of space in the fdt.

Explicitly allocate some extra bytes before to repair this fixup on
lx2162 som.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 board/solidrun/lx2160acex7/lx2160a.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/board/solidrun/lx2160acex7/lx2160a.c b/board/solidrun/lx2160acex7/lx2160a.c
index e0a9e6c51eb..17160d13154 100644
--- a/board/solidrun/lx2160acex7/lx2160a.c
+++ b/board/solidrun/lx2160acex7/lx2160a.c
@@ -81,8 +81,13 @@ int board_fix_fdt(void *fdt)
 	int off = -1, i = 0;
 	u32 is_lx2162 = get_svr() & 0x800;
 
-	if (is_lx2162)
+	if (is_lx2162) {
+		/* allocate space for changes */
+		fdt_increase_size(fdt, 32);
+
+		/* LX2162 does not have second USB controller, disable */
 		do_fixup_by_path_string(fdt, "/usb3@3110000", "status", "disabled");
+	}
 
 	board_fix_fdt_eth(fdt);
 
-- 
2.43.0

